packages/tasit-action/src/contract/Contract.test.js

```diff
@@ -288,94 +285,86 @@ describe("TasitAction.Contract", () => {
           const value = await sampleContract.getValue();
           expect(value).to.equal(rand);
         }
-        confirmationFakeFn();
-      };
+      });
+
+      const errorListener = sinon.fake();
 
       action.on("error", errorListener);
       action.on("confirmation", confirmationListener);
 
-      await mineBlocks(provider, 6);
+      await mineBlocks(provider, 7);
 
-      expect(confirmationFakeFn.callCount).to.equal(6);
-      expect(errorFakeFn.called).to.be.false;
+      expect(confirmationListener.callCount).to.equal(6);
+      expect(errorListener.called).to.be.false;
     });
 
     it("should change contract state and trigger confirmation event - late subscription", async () => {
       action = sampleContract.setValue(rand);
       await action.send();
-
       await action.waitForOneConfirmation();
 
       await mineBlocks(provider, 5);
 
-      const confirmationFakeFn = sinon.fake();
-      const errorFakeFn = sinon.fake();
-
-      const errorListener = message => {
-        errorFakeFn();
-      };
+      const errorListener = sinon.fake();
 
-      const confirmationListener = async message => {
+      const confirmationListener = sinon.fake(async message => {
         const { data } = message;
         const { confirmations } = data;
 
-        if (confirmations == 7) {
+        if (confirmations >= 7) {
           action.off("confirmation");
 
           const value = await sampleContract.getValue();
           expect(value).to.equal(rand);
-
-          confirmationFakeFn();
         }
-      };
+      });
 
       action.on("error", errorListener);
       action.on("confirmation", confirmationListener);
 
       await mineBlocks(provider, 2);
 
-      expect(confirmationFakeFn.called).to.be.true;
-      expect(confirmationFakeFn.callCount).to.equal(1);
-      expect(errorFakeFn.called).to.be.false;
+      // Non-deterministic
+      expect(confirmationListener.callCount).to.be.at.least(1);
```

Comment:
Does the name of this test need to change at all, since it used to be looking for 7 confirmations and now is looking for just one confirmation?

Reply:
This name is `should change contract state and trigger confirmation event - late subscription` I think that is OK since this test creates an action, and start to listening to it after the 6th confirmation (late subscription). The essence of the test case still the same, I've tried to reduce the non-determinism of that.

Reply:
I was wrong about what it's looking for now.

Reply:
And yes, everything you said there is right. Thanks for explaining that. Feel free to resolve it.

Reply:
Ok, resolving this.